sph/include/sph/kernel_ve/density_kern.hpp

```diff
@@ -56,10 +55,10 @@ CUDA_DEVICE_HOST_FUN inline void densityJLoop(int i, T sincIndex, T K, const cst
     T wrho0i = wrho0[i];
     T xmassi = m[i] / rho0i;
 
-    T hInv  = 1.0 / hi;
+    T hInv  = T(1) / hi;
     T h3Inv = hInv * hInv * hInv;
 
-    T kxi      = 0.0;
+    T kxi      = xmassi;
     T whomegai = 0.0;
```

Comment:
You can do the same with whomegai:
whomegai = - T(3) * xmassi

Reply:
same as above